### ISSUE TITLE
issues with Set() + circular references

### ISSUE THREAD

**Issue by zubuzon** [isiahmeadows/thallium#26]
```js

var set1 = new Set();
var set2 = new Set();

        set1.add(set1)
        set2.add(set2)

        strict( set1, set2)

        set1.add(1);
        set2.add(2);
      
    // Should return false, return true
        strict( set1, set2)

```

**Comment by isiahmeadows** [comment]
Check with the latest version that I've just published. This is very similar to a test case I have.

**Comment by zubuzon** [comment]
@isiahmeadows Your test is this: https://github.com/isiahmeadows/thallium/blob/master/test/assertions/match.js#L702

It only does the first "add".  You are missing the last two. So if you copy the test above, the issues still exist.
The issue occur around here: https://github.com/isiahmeadows/thallium/blob/master/match.js#L408

**Comment by isiahmeadows** [comment]
Send a PR with the failing test case, as a separate test.

**Comment by isiahmeadows** [comment]
Fixed in 1db306c.
